### DISCUSSION TITLE
ValueError: min() arg is an empty sequence

### DISCUSSION THREAD

**Discussion by svenborgstrom** [isce-framework/isce2#630] [Q&A] [answered]
<img width="1347" alt="run_08_merge_burst_igram" src="https://user-images.githubusercontent.com/93733824/204126741-8281dbfb-3ab4-4d18-aac9-8e282405f9a9.png">

Good morning everybody,
I'm processing a S1 correlation stack (stackSentinel.py ...) in the run_files folder.
At "run_08_merge_burst_igram" I got the error:

topSwath = min(refSwaths, key = lambda x: x.sensingStart)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: min() arg is an empty sequence
(see also the attached file).

I saw in the forum as well to check if there was a similar post, it was there, but the post was closed without any fix.
About the merge burst step I must also say I previously changed the "stackSentinel.py" script at row 731 into 
"runObj.igram_mergeBurst(acquisitionDates, safe_dict, pairs)" otherwise I couldn't have generated the run_files folder.
Any help to fix my issue would be greatly appreciated, thanks.
Sven.

**Comment by yunjunz** [comment, marked as answer]
I never used the `correlation` workflow. Based on the code, it seems that the [`runObj.burstIgram_mergeBurst()`](https://github.com/isce-framework/isce2/blob/main/contrib/stack/topsStack/stackSentinel.py#L731) is NOT defined anywhere. 

I guess replacing [these lines](https://github.com/isce-framework/isce2/blob/main/contrib/stack/topsStack/stackSentinel.py#L728-L732) from:

```python
    i+=1
    runObj = run()
    runObj.configure(inps, 'run_{:02d}_merge_burst_igram'.format(i))
    runObj.burstIgram_mergeBurst(acquisitionDates, safe_dict, pairs)
    runObj.finalize()
```

to

```python
    i+=1
    runObj = run()
    runObj.configure(inps, 'run_{:02d}_generate_burst_igram'.format(i))
    runObj.generate_burstIgram(acquisitionDates, safe_dict, pairs)
    runObj.finalize()

    i += 1
    runObj = run()
    runObj.configure(inps, 'run_{:02d}_merge_burst_igram'.format(i))
    runObj.igram_mergeBurst(acquisitionDates, safe_dict, pairs)
    runObj.finalize()
```

would fix the issue.

Another way around this is to use the `interferogram` workflow if you are more familiar with it, which includes the correlation products already.

**Reply by svenborgstrom** [reply]
Many thanks Yunjun.
Problem fixed.
Sven.
